tests/infra/ccf.py

```diff
@@ -424,29 +435,40 @@ def vote_using_majority(self, remote_node, proposal_id):
         return res[1]
 
     def vote(self, member_id, remote_node, proposal_id, accept, force_unsigned=False):
-        j_result = self.member_client_rpc_as_json(
-            member_id,
-            remote_node,
-            "vote",
-            f"--proposal-id={proposal_id}",
-            "--accept" if accept else "--reject",
-            "--force-unsigned" if force_unsigned else "",
-        )
-        if j_result.get("error") is not None:
-            return (False, j_result["error"])
+        if os.getenv("HTTP"):
+            script = """
+            tables, changes = ...
+            return true
+            """
+            with remote_node.member_client(member_id) as mc:
+                r = mc.rpc(
+                    "vote", {"ballot": {"text": script}, "id": proposal_id}, signed=True
+                )
+            return (True, r.result)
+        else:
+            j_result = self.member_client_rpc_as_json(
+                member_id,
+                remote_node,
+                "vote",
+                f"--proposal-id={proposal_id}",
+                "--accept" if accept else "--reject",
+                "--force-unsigned" if force_unsigned else "",
+            )
+            if j_result.get("error") is not None:
+                return (False, j_result["error"])
 
         # If the proposal was accepted, wait for it to be globally committed
         # This is particularly useful for the open network proposal to wait
         # until the global hook on the SERVICE table is triggered
         if j_result["result"]:
-            with remote_node.node_client() as mc:
+            with remote_node.node_client(member_id) as mc:
                 wait_for_global_commit(mc, j_result["commit"], j_result["term"], True)
 
         return (True, j_result["result"])
 
     def propose_retire_node(self, member_id, remote_node, node_id):
         return self.propose(
-            member_id, remote_node, "retire_node", f"--node-id={node_id}"
+            member_id, remote_node, "retire_node", None, None, f"--node-id={node_id}"
```

Comment:
minor: the optional arguments could be passed as keyword arguments, since there are some `None`s here and it would be easier to understand what they are

Reply:
Because `propose()` takes a variable number of non-keyword arguments (i.e. `*args` as last parameter), the `script` and `params` arguments cannot be keyword'ed. There are of course ways around this (use of `**kwargs`, etc.) but the variable number of arguments only exists because of our use of the `memberclient` CLI that should disappear soon - I'd rather leave like it for now and add a TODO for when we stop use the `memberclient`.